kubernetes_asyncio/client/configuration.py

```diff
@@ -348,15 +349,18 @@ def logger_format(self, value):
         self.__logger_format = value
         self.logger_formatter = logging.Formatter(self.__logger_format)
 
-    def get_api_key_with_prefix(self, identifier, alias=None):
+    async def get_api_key_with_prefix(self, identifier, alias=None):
```

Comment:
Unfortunately this file is auto-generated and it's better not to change it. 

There is a mechanism to refresh GCP token:

https://github.com/tomplus/kubernetes_asyncio/blob/master/kubernetes_asyncio/config/kube_config.py#L611

Example:

https://github.com/tomplus/kubernetes_asyncio/blob/master/examples/tail.py#L75 

Could you try to extend it?

Reply:
I have extended the `refresh_token` method used for GCP tokens for exec credentials. 

While I was developing this change, I ran into an issue that I suspect may also be impacting the refreshing of GCP tokens. I have confirmed that this change works when `refresh_token` is passed a configuration object. However, there is a problem when `refresh_token` is called without a configuration object. 

This problem is that when no configuration object is specified a local configuration object is initialized and `refresh_token` updates the `api_key` attribute of the local object. In this case, there is no way to have the `api_client` access the updated `api_key`. 

Unless I am not understanding something about GCP tokens, they would run into the same problem in this example https://github.com/tomplus/kubernetes_asyncio/blob/master/examples/tail.py as `refresh_token` is called without a configuration object as a parameter. I have not confirmed that this is the case as I do not have access to an environment to test GCP tokens.

I am curious what you think about this and how you would like to proceed.

Reply:
Sounds good. I also thought about how to fix this issue and I do not think it's possible without changing generated files. 

I updated the `tail.py` example and changed `refresh_token` to raise `NotImplementedError` when a configuration is not passed.